sbi/simulators/simutils.py

```diff
@@ -117,56 +117,59 @@ def simulate_in_batches(
     parameter_sample_fn: Callable,
     num_samples: int,
     simulation_batch_size: int,
-    x_dim: torch.Size,
 ) -> (torch.Tensor, torch.Tensor):
     """
-    Draw `num_samples` parameter sets and simulate them in batches of size `simulation_batch_size`.
+    Draw `num_samples` parameter sets and simulate them in batches of size
+     `simulation_batch_size`.
 
     Features: - allows to simulate in batches of arbitrary size
               - if `simulation_batch_size==-1`, all simulations are run at the same time
               - simulator output can be np.array or torch.Tensor
 
     Args:
-        simulator: simulator function.
-            If `simulation_batch_size == 1`: takes in parameters of shape (1, num_dim_parameters)
-                and outputs xs of shape (1, num_dim_x)
-            If `simulation_batch_size > 1`: takes in thetas of shape (simulation_batch_size, num_dim_parameters)
-                and outputs xs of shape (simulation_batch_size, num_dim_x)
+        simulator: simulator function that takes in parameters theta and outputs data x
+            If `simulation_batch_size == 1`: takes in parameters of shape
+             (1, num_dim_parameters) and outputs xs of shape (1, num_dim_x)
+            If `simulation_batch_size > 1`: takes in thetas of shape
```

Comment:
1) I think using len(parameters) would be confusing: parameters is a tensor of shape (N, num_dim_parameters), so len(parameters)=N. I understand that what you mean is the length of a single parameter set, but this variable does not exist in the code and I think it will be confusing for users. Same for len(x).

2) yes, they can be grouped

Reply:
About 1., just to be sure: did you consider your answer in the light of the last sentence?

Reply:
I did. But now re-reading it, what you might have meant is:
(1, num_dim_x) --> numel(x)
If this case, I agree, we can do that, and I actually like it. I had interpreted your previous message as:
(1, num_dim_x) --> (1, numel(x))
which I referred my comment to.

Reply:
`numel` is well defined: it's the product of the components of the shape. I think I am confused by the use of `dim` and `num_dim` in several spots, so perhaps could you rather explain to me if with `num_dim_x` one means the length of the shape/size iterable (number of entries). This is the customary definition in numpy/PyTorch. I might be missing the point...

Reply:
Ah, no. When I mean that, I write .ndim(), but I can see how this is confusing.

I use num_dim for what you would probably call dim. E.g. a parameter set [1., 2., 3.] would have num_dim=3.

Reply:
would advocate for using the standard nomenclature of numpy/pytorch; and document any departures extensively. A parameter 'set' (rather a list or iterable) [1, 2, 3] would have dimension 1, ndim 1, shape (1,) length 3.

Reply:
Fine by me.